### PR TITLE
Cron expr conversion

### PR DESCRIPTION
**Pull request by rahu-eGov**
Converting cron expression to calculate fiscal event search time period.

### CODE REVIEW

File: reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java [hunk 1]
```diff
@@ -87,16 +91,40 @@ public String getFiscalEventDetailsInformation(List<FiscalEvent> fiscalEventList
...
+            Long overlapTime = Long.parseLong(plainTime) * 60 * 1000;
+
+            String cronExpression = applicationConfiguration.getSampleJobFrequency();
+            String[] fields = StringUtils.tokenizeToStringArray(cronExpression, " ");
+
+            if (fields == null || fields.length != 6) {
+                log.error(String.format(">>>>> Cron expression must consist of 6 fields (found %d in \"%s\")",
+                        fields.length, cronExpression));
+
+                throw new GenericCustomException("CRON_EXPRESSION",
```
Comment (rushang7-eGov) [inline reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java:115]: Instead of throwing an error, can we just drop the last part and move ahead. Just add a log statement that considering only the first 6 parts and dropping the 7th element.

File: reference-adapter/mgramseva-ifix-adapter/src/main/resources/application.properties [hunk 2]
```diff
@@ -68,7 +68,7 @@ ifix.department.entity.search.path=/departmentEntity/v1/_search
 pspcl.ifix.event.receiver.name=mgramseva-qa
 
 #Value in seconds
-ifix.fiscal.event.search.time.interval.minutes=10
+ifix.fiscal.event.search.time.overlap.minutes=15
```
Comment (rushang7-eGov) [inline reference-adapter/mgramseva-ifix-adapter/src/main/resources/application.properties:71]: Don't add this in the helm chart. Let this just be here. 
Remove the old variable from the helm chart and env file.

File: reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java (hunk 1 above)
Comment (rahu-eGov) [reply reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java:115]: I have written for 6 and more than six and working fine but while testing other scenarios come like empty, null and less than 6 - all these scenario throws exception.

`if (fields == null || fields.length < 6) {
            log.error(String.format("Cron expression length is not valid"));
            throw new GenericCustomException("CRON_EXPRESSION", "Cron expression length is not valid");
        }`
Should I put that condition and throw exception.

Comment (rushang7-eGov) [reply reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java:115]: There won't be null, empty string. In those scenarios, the quartz cron job will also fail. No need to handle those.

Comment (rushang7-eGov) [reply reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java:115]: Don't throw any error. Just log that you are dropping the 7th element.

Comment (rahu-eGov) [reply reference-adapter/mgramseva-ifix-adapter/src/main/java/org/egov/ifix/utils/DataWrapper.java]: Done

File: reference-adapter/mgramseva-ifix-adapter/src/main/resources/application.properties (hunk 2 above)
Comment (rahu-eGov) [reply reference-adapter/mgramseva-ifix-adapter/src/main/resources/application.properties:71]: will raise PR in DevOp repo

**Review by rushang7-eGov — APPROVED** [review]
